zilencer/views.py

```diff
@@ -166,10 +187,19 @@ def remote_server_notify_push(request: HttpRequest, entity: Union[UserProfile, R
         server=server,
     ))
 
+    encrypted_android_data, encrypted_apple_data = prepare_encrypted_data(android_devices + apple_devices,
+                                                                          encrypted_payloads)
+
+    android_devices = [d for d in android_devices if not d.encrypted]
```

Comment:
This new value has a quite different meaning from the old value called `android_devices`. It should therefore have a different name which expresses that different meaning.

(Same for `apple_devices` below.)

Alternatively, this list comprehension could go directly in the argument to the function call, rather than give it a name at all.